src/memory_pool.c

```diff
@@ -15,6 +15,30 @@
 #include <assert.h>
 #include <stdlib.h>
 
+enum umf_result_t
+umfPoolCreateEx(const struct umf_memory_pool_ops_t *pool_ops, void *pool_params,
+                const struct umf_memory_provider_ops_t *provider_ops,
+                void *provider_params, umf_memory_pool_handle_t *hPool) {
+    umf_memory_provider_handle_t provider = NULL;
```

Comment:
Check for pointers != null

Reply:
I've added a null-check for the output handle pointer and asserts after provider/pool creation.
From what I've seen in `umfPoolCreate` it null-checks only the input provider handle, whereas `umfMemoryProviderCreate` doesn't check any parameters. Should we validate `...provider/pool_ops_t` pointers and input/output handle pointers in those functions?

Reply:
Please validate all input pointers. About other functions - we should add similar checks there, probably in separate PRs. Please note that we'll have to do it anyway when we enable coverity :)

Reply:
Done, I've omitted the params, since pool/provider implementations not necessarily need a parameter for its initialization.